src/data/number/numberFn.ts

```diff
@@ -1,10 +1,90 @@
 import { assert, assertInteger } from '../../utils/assert'
-import { plString } from '../string/stringFn'
 import { PLNumber } from './PLNumber'
 
-export const plNumber = (value: number): PLNumber => new PLNumber(value)
+export interface DecimalResult {
+  intValue: number
+  decimals: number
+}
+
+export function plFloat(x: number): PLNumber {
```

Comment:
`plFloat` is looks like a new type and it is missleading for PL devs and for PL users as well.

I recommend to use the `plNumber` and overload it.

you need to refactore it a bit like this:
```
export function plNumber(value: string): PLNumber
export function plNumber(value: number, decimal: number): PLNumber
export function plNumber(value: number | string, decimal = 0): PLNumber {
  if (typeof value === 'string') {
  assertNumeric(strValue)
     const decimalObj = isScientific(strValue) ? parseScientificString(strValue) :  parseDecimalString(strValue)
     return plNumber(decimalObj.intValue, decimalObj.decimals)
  } else {
      assertInteger(intValue)
      assertInteger(decimals)
      return new PLNumber(intValue, decimals)
  }
}

Reply:
Done.